giskard/rag/knowledge_base_testset_generator.py

```diff
@@ -0,0 +1,137 @@
+from typing import Sequence
+
+import numpy as np
+import pandas as pd
+
+from ..datasets import Dataset
+from ..llm.errors import LLMGenerationError
+from ..llm.generators import BaseDataGenerator
+from .embeddings import EmbeddingsBase, OpenAIEmbeddings
+from .prompts import ANSWER_GENERATION_PROMPT, QUESTION_GENERATION_PROMPT
+from .vector_store import VectorStore
+
+
+class KnowledgeBaseTestsetGenerator(BaseDataGenerator):
+    _question_generation_prompt = QUESTION_GENERATION_PROMPT
+    _answer_generation_prompt = ANSWER_GENERATION_PROMPT
+    _difficulty_level = 1
+
+    def __init__(
+        self,
+        knowledge_df,
+        model_name: str,
+        model_description: str,
+        context_neighbors: int = 4,
+        context_similarity_threshold: float = 0.2,
+        context_window_length: int = 8192,
+        embedding_model: EmbeddingsBase = None,
+        language: str = "english",
```

Comment:
Not clear what code you are using for language. Maybe check IETF BCP 47? It's common to use "en", "fr", etc.

Reply:
Yes let's keep [ISO 639-1](https://en.wikipedia.org/wiki/ISO_639-1) as we did for [GSK-2152].